File: Python/145-binary-tree-postorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def postorderTraversal(self, root: TreeNode) -> List[int]:
        if not root:
            return []
        stack = [root]
        res = []
        while stack:
            curNode = stack.pop()
            res.append(curNode.val)
            if curNode.left: stack.append(curNode.left)
            if curNode.right: stack.append(curNode.right)
        res.reverse()
        return res

File: Go/145-binary-tree-postorder-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func reverse(nums []int) []int{
    i, j := 0, len(nums) - 1
    for i < j {
        nums[i], nums[j] = nums[j], nums[i]
        i += 1
        j -= 1
    }
    return nums
}

func postorderTraversal(root *TreeNode) []int {
    if root == nil {
        return make([]int, 0)
    }
    stack := []*TreeNode{root}
    var res []int
    for len(stack) != 0 {
        cur := stack[len(stack) - 1]
        stack = stack[:len(stack) - 1]
        res = append(res, cur.Val)
        if cur.Left != nil {
            stack = append(stack, cur.Left)
        }
        if cur.Right != nil {
            stack = append(stack, cur.Right)
        }
    }
    return reverse(res)
}